TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/integ_test/integ_test/ma_workflow_test.py

```diff
@@ -1,102 +1,49 @@
 
 
-def run_on_all_cases(test_cases: List, operation: Callable) -> None:
-    for tc in test_cases:
-        operation(tc)
+def record_test(test_case: MATestBase, record_data) -> None:
+    record_data({"name": test_case.__class__.__name__, "description": test_case.description})
 
 
-# The test_cases parameter here is dynamically provided by the pytest_generate_tests() function in conftest.py. This
-# function will add a parametrize tag on this test to provide the 'test_cases' it has collected
-def test_migration_assistant_workflow(record_data, test_cases: List[MATestBase]):
-    logger.info(f"Performing the following test cases: {test_cases}")
-    record_tests(test_cases=test_cases, record_data=record_data)
-    control_test_case = test_cases[0]
+# The test_case parameter here is dynamically provided by the pytest_generate_tests() function in conftest.py. This
+# function will add a parametrize tag on this test to provide the 'test_case' it has collected
+def test_migration_assistant_workflow(record_data, test_case: MATestBase):
+    logger.info(f"Performing the following test case: {test_case}")
+    record_test(test_case=test_case, record_data=record_data)
 
     # Enable for stepping through workflows with Python debugger
     #breakpoint()
 
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.test_before())
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.snapshot_before())
-    control_test_case.snapshot_create()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.snapshot_after())
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.metadata_before())
-    control_test_case.metadata_migrate()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.metadata_after())
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.backfill_before())
-    control_test_case.backfill_start()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.backfill_during())
-    control_test_case.backfill_wait_for_stop()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.backfill_after())
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.replay_before())
-    control_test_case.replay_start()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.replay_during())
-    control_test_case.replay_wait_for_stop()
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.replay_after())
-    run_on_all_cases(test_cases=test_cases, operation=lambda case: case.test_after())
+    test_case.test_before()
+    test_case.prepare_workflow_parameters()
+    test_case.workflow_start()
+    test_case.workflow_setup_clusters()
+    test_case.prepare_clusters()
+    test_case.workflow_perform_migrations()
+    test_case.verify_clusters()
+    test_case.workflow_finish()
+    test_case.test_after()
```

Comment:
looking at this & the implementations, it feels like writing these workflows entirely within the argo TS builders could be really nice.

Reply:
Yeah it seems like a good area to try out the argo TS building process